### DISCUSSION TITLE
Option to set default query function for useQuery

### DISCUSSION THREAD

**Discussion by sharmapukar217** [rakkasjs/rakkasjs#74] [Q&A] [answered]
Is there any way to pass default queryFn for useQuery ?

**Comment by cyco130** [comment, marked as answer]
No. Actually I left it out on purpose because it would force me to make the query function argument optional in TypeScript, and things will blow up in runtime if you don't give a default query function. I think it was a bad design decision in `react-query` too. May I suggest you use a wrapper hook like the following?

```ts
function useDefaultQuery(key, options) {
  return useQuery(key, defaultQueryFn, options);
}
```

**Reply by sharmapukar217** [reply]
> ```ts
> function useDefaultQuery(key, options) {
>   return useQuery(key, defaultQueryFn, options);
> }
> ```
This is exactly the same which I'm using currently 😄
